edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/CredentialsCache.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Hub.Core
+{
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+
+    public class CredentialsCache : ICredentialsCache
+    {
+        readonly IDictionary<string, IClientCredentials> cache = new ConcurrentDictionary<string, IClientCredentials>();
```

Comment:
Does this need to be a ConcurrentDictionary, if it's always guarded by a lock?

Reply:
ah, I wanted to revert that.. (and thought I did). Will fix.